features/nanostack/mbed-mesh-api/source/NanostackEthernetInterface.cpp

```diff
@@ -97,8 +97,26 @@ nsapi_error_t NanostackEthernetInterface::do_initialize()
 
 nsapi_error_t Nanostack::EthernetInterface::bringdown()
 {
-    if (enet_tasklet_disconnect(true)) {
+    nanostack_lock();
+    int8_t status = enet_tasklet_disconnect(true);
+    nanostack_unlock();
+
+    if (status == -1) {
+        return NSAPI_ERROR_DEVICE_ERROR;
+    } else if (status == -2) {
+        return NSAPI_ERROR_NO_MEMORY;
+    } else if (status == -3) {
+        return NSAPI_ERROR_ALREADY;
+    } else if (status != 0) {
         return NSAPI_ERROR_DEVICE_ERROR;
     }
+
+    if (_blocking) {
+        int32_t count = disconnect_semaphore.wait(30000);
```

Comment:
30s timeout?

Reply:
I took it from the bringup procedure... Is there a specified timeout that you'd suggest?